jobs/webcompat-kb/webcompat_kb/update_schema.py

```diff
@@ -757,6 +757,131 @@ def check_templates() -> None:
         raise
 
 
+def update_schema_if_needed(
+    client: BigQuery,
+    schema_path: os.PathLike,
+    stage: bool,
+    recreate: bool,
+    delete_extra: bool,
+) -> None:
+    schema_path = os.path.abspath(schema_path)
+
+    metadata_dataset = DatasetId(client.project_id, "metadata")
+    if stage:
+        metadata_dataset = stage_dataset(metadata_dataset)
+
+    src_hash = get_templates_hash(schema_path).hex()
+    last_update_time, last_update_hash = get_last_update(client, metadata_dataset)
+
+    logging.info(f"Templates have hash {src_hash}")
+    logging.info(f"Deployed schema have hash {last_update_hash}")
+
+    update_needed = (
+        recreate
+        or last_update_hash != src_hash
+        or (last_update_time and last_update_time.date() < datetime.now().date())
+    )
+    if not update_needed:
+        logging.info("No changes to deploy")
+        return
+
+    templates_by_dataset = load_templates(client.project_id, schema_path)
+    if not lint_templates(templates_by_dataset):
+        raise ValueError("Template lint failed")
+
+    dataset_mapping = {dataset.id: dataset.id for dataset in templates_by_dataset}
+    rewrite_tables = set()
+    if stage:
+        dataset_mapping = {
+            dataset: stage_dataset(dataset) for dataset in dataset_mapping
+        }
+        # If a table is in the target dataset, use that, otherwise reuse the
+        # table in the source dataset. This is because we don't always have
+        # copies of the tables in the _test datasets for various reasons.
+        for dataset, target_dataset in dataset_mapping.items():
+            rewrite_tables |= set(
+                SchemaId(client.project_id, dataset.dataset, item.table_id)
+                for item in client.client.list_tables(target_dataset.dataset)
+                if item.table_type != "VIEW"
+            )
+        logging.debug("\n".join(str(item) for item in rewrite_tables))
+
+    schema_id_mapper = SchemaIdMapper(dataset_mapping, rewrite_tables)
+    update_schemas(
+        client,
+        templates_by_dataset,
+        dataset_mapping,
+        schema_id_mapper,
+        delete_extra,
+    )
+    record_update(client, metadata_dataset, src_hash)
+
+
+class UpdateStagingData(EtlJob):
+    name = "update-staging-data"
+    default = False
+
+    @classmethod
+    def add_arguments(cls, parser: argparse.ArgumentParser) -> None:
+        group = parser.add_argument_group(
+            title="Update Staging Data", description="update-staging-data arguments"
+        )
+        group.add_argument(
+            "--update-staging-data-views",
+            action="store_true",
+            default=False,
+            help="Redeploy views to staging",
+        )
+
+    def default_dataset(self, args: argparse.Namespace) -> str:
+        return args.bq_kb_dataset
```

Comment:
Not super important, but wonder if `bq_kb_dataset` should be a required argument as it was in the `get_parser_create_test_dataset` ( I see there is a check `assert args.bq_kb_dataset is not None`)?